app/views/whitehall_migration/documents.html.erb

```diff
@@ -1,6 +1,38 @@
-<h1>Documents</h1>
+<% content_for :back_link, render_back_link(href: whitehall_migration_path(@whitehall_migration)) %>
+<% content_for :title, "Document Imports" %>
 
-<% @documents.each do |document| %>
-  <li>ID: <%= document.id %></li>
-  <li>Status: <%= document.state %></li>
-<% end %>
+<% states = ["imported", "sync_failed", "completed"] %>
+
+<% items = @documents.map do |document_import|
+    {
+      link: {
+        text: states.include?(document_import.state) ? document_import.document.current_edition.title : document_import.id.to_s,
+        path: whitehall_migration_document_path(document_import.whitehall_migration_id, document_import),
+      },
+      metadata: {
+        status: document_import.state.to_s,
+        error_message: document_import.error_log ? "Error log: #{document_import.error_log}" : ""
+      },
+      subtext: states.include?(document_import.state) ? "View: <a href=\"#{Plek.find('whitehall-admin')}/government/admin/by-content-id/#{document_import.content_id}\">Whitehall</a> View: <a href=\"#{Plek.find('content-publisher')}/documents/#{document_import.content_id}:en\">Content Publisher</a>".html_safe : "View: <a href=\"#{Plek.find('whitehall-admin')}/government/admin/by-content-id/#{document_import.content_id}\">Whitehall</a>".html_safe,
+    }
+  end
+%>
+
+<%= render "govuk_publishing_components/components/document_list", items: items %>
```

Comment:
Do we want a fallback for when there are no items?

Reply:
changes to `items ||= []`